### PR TITLE
Tools-2350 Tools-2424 add stop write size to info set

### PR DESCRIPTION
**Pull request by jdogmcsteezy**

### CODE REVIEW

File: lib/view/templates.py [hunk 1]
```diff
@@ -69,24 +69,44 @@ def project_xdr_req_shipped_errors(s, esc, ess):
...
+    """
+    values_total = 0.0
+    weighted_total = 0.0
 
     for w, v in zip(weights, values):
-        weighted_avg += w * v
-        total += v
+        weighted_value = w * v
+        weighted_total += weighted_value
+        values_total += v
```
Comment (dwelch-spike) [inline lib/view/templates.py:104]: I thought weighted averages divide by the sum of the weights in the end. Please double check that this is correct.

File: lib/view/templates.py [hunk 2]
```diff
@@ -505,6 +569,49 @@ def set_index_projector(enable_index, index_populating):
...
+                ),
+                Field(
+                    "Used%",
+                    Projectors.Div(
+                        Projectors.Sum(
+                            Projectors.Number(
+                                "set_stats", "memory_data_bytes", "n-bytes-memory"
+                            ),
+                            Projectors.Number(
+                                "set_stats", "device_data_bytes", "n-bytes-memory"
```
Comment (dwelch-spike) [inline lib/view/templates.py:592]: should this be "n-bytes-device"?

File: lib/view/templates.py [hunk 3]
```diff
@@ -1458,8 +1560,8 @@ def numeric_sum_aggregator_selector(key, is_numeric):
 
 
 def latency_weighted_avg(edatas: list[EntryData]):
-    pcts = map(lambda edata: edata.value, edatas)
-    ops_sec_values = map(lambda edata: edata.record["ops/sec"], edatas)
+    ops_sec_values: map[float] = map(lambda edata: edata.record["ops/sec"], edatas)
```
Comment (dwelch-spike) [inline lib/view/templates.py:1563]: is this a valid type hint? isn't the result of map() a list?

File: lib/view/templates.py [hunk 4]
```diff
@@ -263,67 +290,103 @@ def usage_weighted_avg(edatas: list[EntryData]):
...
             ),
         ),
         Subgroup(
             "Memory",
             (
+                Field(
+                    "Total",
+                    Projectors.Number("ns_stats", "memory-size", "total-bytes-memory"),
+                    hidden=True,
+                ),
```
Comment (dwelch-spike) [inline lib/view/templates.py:336]: Why is this hidden?

**Review by dwelch-spike — CHANGES_REQUESTED** [review]
A couple questions, mostly want to make sure the averaging is correct.

File: lib/view/templates.py [hunk 5]
```diff
@@ -69,24 +69,44 @@ def project_xdr_req_shipped_errors(s, esc, ess):
...
 
+def _ignore_zero(num: int):
+    if num == 0:
+        return None
+
+    return num
+
+
+def _ignore_null(s: str):
+    if s == "null":
```
[the remark this answers is not in the document]
Comment (jdogmcsteezy) [reply lib/view/templates.py:80]: We do not but I can make it that way.

File: lib/view/templates.py (hunk 1 above)
Comment (jdogmcsteezy) [reply lib/view/templates.py:104]: renamed the variables to match the equation you saw

File: lib/view/templates.py (hunk 2 above)
Comment (jdogmcsteezy) [reply lib/view/templates.py:592]: good catch

File: lib/view/templates.py (hunk 3 above)
Comment (jdogmcsteezy) [reply lib/view/templates.py:1563]: It is valid. A map returns a generator like object called a map

File: lib/view/templates.py (hunk 4 above)
Comment (jdogmcsteezy) [reply lib/view/templates.py:336]: So that it is not displayed. I am following precedence here. If somebody wants the total they can open a ticket

Comment (jdogmcsteezy) [reply lib/view/templates.py:336]: The value is needed for the weighted average calculation.  That does not mean it should be displayed

File: lib/view/templates.py (hunk 1 above)
Comment (jdogmcsteezy) [reply lib/view/templates.py:104]: That's what happens when you derive the equation yourself haha.  I was thinking the pct should be considered the weight and the bytes should be considered the values but in more abstract terms the values should be the same unit of the result you are expecting

**Review by dwelch-spike — APPROVED** [review]
LGTM

**Pull request merged**